hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicRootedOzoneFileSystem.java

```diff
@@ -766,14 +766,22 @@ private void deleteBucketFromVolume(Path f, OFSPath ofsPath)
       throws IOException {
     OzoneVolume volume =
         adapterImpl.getObjectStore().getVolume(ofsPath.getVolumeName());
-    try {
-      volume.deleteBucket(ofsPath.getBucketName());
-    } catch (OMException ex) {
-      // bucket is not empty
-      if (ex.getResult() == BUCKET_NOT_EMPTY) {
-        throw new PathIsNotEmptyDirectoryException(f.toString());
-      } else {
-        throw ex;
+    int retryCount = 3;
+    int count = 0;
+    while (count < retryCount) {
```

Comment:
might be also better to add a small comment explaining why we retry here as a new reader can get confused about why the retry is being performed.

Reply:
Why do we need to retry `deleteBucket`?  Is the goal to fix intermittent "bucket not empty" problem (e.g. HDDS-8526, HDDS-8912)?  If so, I think it should be fixed separately.

Reply:
Yes "bucket not empty" is intermittent because of key is deleted but still not flushed to DB yet. So in between if we check bucket empty which doesn't give proper result [isKeyPresentInTable](https://github.com/apache/ozone/blob/92b49ebde98774f9f63ad091bce662d20c95096f/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataManagerImpl.java#L1030).
Even though in this method we are checking both cache and table, but doesn't ensure it will give exact information about bucket empty when key was deleted just before this operation check.
One way is to retry or other way is to flush cache into table and then check isBucketEmpty.

Reply:
Removed this change from this PR.

Reply:
This is now fixed in [HDDS-8526](https://issues.apache.org/jira/browse/HDDS-8526)